app/src/components/ArtifactTimeline.tsx

```diff
@@ -0,0 +1,102 @@
+import { Center, ScrollArea, Space, Text, Timeline } from '@mantine/core';
+import { Message } from 'tabler-icons-react';
+import { useEffect, useState } from 'react';
+import ArtifactComments from './ArtifactComments';
+import { ArtifactCommentProps } from './ArtifactComments';
+
+interface ArtifactTimelineProps {
+  extensions?: fhir4.Extension[];
+}
+
+type commentProps = {
+  author?: string;
+  date?: string;
+  body?: string;
+  type?: string;
+};
+
+function addArtifactComment({ date, body, author }: ArtifactCommentProps) {
+  return (
+    <div>
+      <ArtifactComments date={date} body={body} author={author} />
+    </div>
+  );
+}
+
+function noCommentsAvailable() {
+  return (
+    <div>
+      <Center>
+        <Text color="red">
+          <i>No Comments Found for Artifact</i>{' '}
+        </Text>
+      </Center>
+    </div>
+  );
+}
+
+export default function ArtifactTimeline(extensionArr: ArtifactTimelineProps) {
+  const [height, setWindowHeight] = useState(0);
+
+  useEffect(() => {
+    const handleResize = () => {
+      setWindowHeight(window.innerHeight);
+    };
+    handleResize();
+    window.addEventListener('resize', handleResize);
+    return window.removeEventListener('resize', handleResize);
+  }, []);
+
+  if (extensionArr?.extensions) {
+    if (extensionArr.extensions.length === 0) {
+      return <div>{noCommentsAvailable()}</div>;
+    } else {
```

Comment:
I don't think we need this if statement. `extensionArr.extensions.forEach` won't add anything to `commentArray` if `extensionArry.extensions.length === 0`, so it will reach the `if (commentArray.length == 0)` statement and return the `noCommentsAvailable` message then.

Reply:
Good point! Ill get rid of that unnecessary check. I still think the `noCommentsAvailable` function should stay, however, cause there are two instances where there could be no comments. One where the artifact doesn't have an extension array in the first place and one in which it does have an extension array but it doesn't have any actual comments.